src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private int randomNumber = random.nextInt(5);
```

Comment:
The random number should be generated inside the `getRandomFigure` method to ensure a new random figure is generated each time the method is called. Consider moving `randomNumber` to be a local variable inside `getRandomFigure`.